lib/signalfx/lambda/tracing.rb

```diff
@@ -8,103 +8,75 @@ module Lambda
-        }
-
-        tags = tags.merge(tags_from_arn(context.invoked_function_arn))
-      end
+          response = yield event: event, context: context
+          scope.span.set_tag("http.status_code", response[:statusCode]) if response[:statusCode]
 
-      def self.tags_from_arn(arn)
-        _, _, _, region, account_id, resource_type, _, qualifier = arn.split(':')
+          response
+        rescue => error
+          if scope
+            scope.span.set_tag("error", true)
+            scope.span.log_kv(key: "message", value: error.message)
+          end
 
-        tags = {
-          'aws_region' => region,
-          'aws_account_id' => account_id,
-        }
+          # pass this error up
+          raise
+        ensure
+          scope.close if scope
 
-        if qualifier
-          case resource_type
-          when 'function'
-            tags['aws_function_qualifier'] = qualifier
-          when 'event-source-mappings'
-            tags['event_source_mappings'] = qualifier
-          end
+          # flush the spans before leaving the execution context
+          @reporter.flush
         end
 
-        tags
-      end
+        def wrapped_handler(event:, context:)
+          wrap_function(event, context, &@handler)
+        end
 
-      def self.register_handler(&handler)
-        @handler = handler
-      end
+        def register_handler(&handler)
+          @handler = handler
+        end
 
-      def self.init_tracer(event)
-        access_token = ENV['SIGNALFX_ACCESS_TOKEN']
-        ingest_url = ENV['SIGNALFX_INGEST_URL'] || 'https://ingest.signalfx.com/v1/trace'
-        service_name = ENV['SIGNALFX_SERVICE_NAME'] || event.function_name
-        @span_prefix = ENV['SIGNALFX_SPAN_PREFIX'] || 'lambda_ruby_'
-
-        # configure the trace reporter
-        headers = { }
-        headers['X-SF-Token'] = access_token if !access_token.empty?
-        encoder = Jaeger::Client::Encoders::ThriftEncoder.new(service_name: service_name)
-        sender = Jaeger::Client::HttpSender.new(url: ingest_url, headers: headers, encoder: encoder, logger: Logger.new(STDOUT))
-        @reporter = Jaeger::Client::Reporters::RemoteReporter.new(sender: sender, flush_interval: 1)
-
-        # propagation format configuration
-        injectors = {
-          OpenTracing::FORMAT_TEXT_MAP => [Jaeger::Client::Injectors::B3RackCodec]
-        }
-        extractors = {
-          OpenTracing::FORMAT_TEXT_MAP => [SignalFx::Lambda::Tracing::B3TextMapCodec]
-        }
-
-        OpenTracing.global_tracer = Jaeger::Client.build(
-          service_name: service_name,
-          reporter: @reporter,
-          injectors: injectors,
-          extractors: extractors
-        )
-
-        @tracer = OpenTracing.global_tracer
+        def init_tracer(event)
+          access_token = ENV['SIGNALFX_ACCESS_TOKEN']
+          ingest_url = ENV['SIGNALFX_TRACING_URL'] || 'https://ingest.signalfx.com/v1/trace'
```

Comment:
I would also have a unified URL setting that can be used for both tracing and metrics.  Maybe call it `SIGNALFX_ENDPOINT_URL` or something so that it can apply to gateway and ingest generically.

Reply:
But definitely also keep the ability to split up where metrics and traces go separately.